src/sections/create-collection/useGetAllFacetableMetadataFields.ts

```diff
@@ -0,0 +1,50 @@
+import { useEffect, useState } from 'react'
+import { MetadataBlockInfoRepository } from '../../metadata-block-info/domain/repositories/MetadataBlockInfoRepository'
+import { MetadataField } from '../../metadata-block-info/domain/models/MetadataBlockInfo'
+import { getAllFacetableMetadataFields } from '../../metadata-block-info/domain/useCases/getAllFacetableMetadataFields'
+
+interface Props {
+  metadataBlockInfoRepository: MetadataBlockInfoRepository
+}
+
+interface UseGetAllFacetableMetadataFieldsReturn {
+  facetableMetadataFields: MetadataField[]
+  error: string | null
+  isLoading: boolean
+}
+
+export const useGetAllFacetableMetadataFields = ({
+  metadataBlockInfoRepository
+}: Props): UseGetAllFacetableMetadataFieldsReturn => {
+  const [facetableMetadataFields, setFacetableMetadataFields] = useState<MetadataField[]>([])
+  const [isLoading, setIsLoading] = useState<boolean>(true)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    const handleGetAllFacetableMetadataFields = async () => {
+      setIsLoading(true)
+      try {
+        const facetableMetadataFieldsResponse: MetadataField[] =
+          await getAllFacetableMetadataFields(metadataBlockInfoRepository)
+
+        setFacetableMetadataFields(facetableMetadataFieldsResponse)
+      } catch (err) {
+        const errorMessage =
+          err instanceof Error && err.message
+            ? err.message
+            : 'Something went wrong getting all the facetable metadata fields. Try again later.'
+        setError(errorMessage)
```

Comment:
CreateCollection.tsx, which is using this hook, doesn't read the error field.  So if an error occurs, it looks like the page will be stuck in loading mode, without any feedback about the error.  Could we print the error message to the console, so that the error message can be seen by developers? Or will the error property be handled in a future code change that includes more general error handling?

Reply:
I made an error handling logic to show any error related to data loading.